Pegasus/include/Math.hpp

```diff
@@ -18,11 +18,87 @@
 #include <set>
 #include <unordered_map>
 #include <unordered_set>
+#include <numeric>
 
 namespace pegasus
 {
 namespace math
 {
+namespace fp
+{
+constexpr double g_floatingPointThreshold = 1e-4;
```

Comment:
@ilia-glushchenko well, I don't have any data, but I would use something closer to 1e-6 or 1e-8. The best way would be to use `std::numeric_limits<double>::epsilon` and then scale this to our numbers (something like the first answer [here](https://stackoverflow.com/questions/4010240/comparing-doubles)), but I suppose that's too slow for our purposes.